Plugins/GitUIPluginInterfaces/ObjectId.cs

```diff
@@ -66,7 +80,7 @@ public static ObjectId Random()
         [MustUseReturnValue]
         public static ObjectId Parse(string s)
         {
-            if (s?.Length is not Sha1CharCount || !TryParse(s.AsSpan(), out ObjectId id))
+            if ((s?.Length is not Sha1CharCount && s?.Length is not Sha256CharCount) || !TryParse(s.AsSpan(), out ObjectId id))
```

Comment:
Merging the patterns might produce better code.

```suggestion
            if (s?.Length is not (Sha1CharCount or Sha256CharCount) || !TryParse(s.AsSpan(), out ObjectId id))
```